src/arith/bound_deducer.cc

```diff
@@ -85,21 +85,24 @@ class BoundDeducer : public ExprVisitor {
   void VisitExpr(const PrimExpr& e) final {
     if (!success_) return;
     if (iter_ < path_.size() && e.get() == path_[iter_++]) {
-      ExprVisitor::VisitExpr(e);
+      if (const AddNode* op = e.as<AddNode>()) {
```

Comment:
Manually expansion here is not as effective as number of Expr grows. I know that it is a bit hard to hit to default case. Another way would be to simply use ExprFunctor(which defaults to failure) then dispatch again.